src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -79,6 +81,18 @@ constexpr auto housekeeping_jit = 5ms;
 
 namespace archival {
 
+bool is_shutdown_exception(const ss::nested_exception& ex) {
```

Comment:
nit: `is_nested_shutdown_exception()` or something else? Should this be placed in the `ssx` namespace for future use elsewhere instead of a free function in this file?

Reply:
I don't think this should be in `ssx` because it's not clear if someone wants `&&` or `||` the condition. TBH I'd probably just inline this function.

Reply:
Also please put in an anonymous namespace to prevent symbol clashes.

Reply:
it has internal linkage so no problem with symbol clashes

Reply:
> it has internal linkage so no problem with symbol clashes

It's not marked static or in an anonymous namespace it doesn't have internal linkage?

Also sorry to not be clear, when I said `inline` I didn't mean the keyword, but to actually just remove the function and put the condition directly into the `if` condition.

Reply:
right, changed inline to static